particle/converters/bimap.py

```diff
@@ -100,3 +103,99 @@ def __str__(self):
     def __len__(self):
         """Returns the number of matches."""
         return len(self._to_map)
+
+
+def DirectionalMaps(name_A, name_B, converters=(str, str), filename=None):
+        """
+        Directional map class providing a to and from mapping.
+
+        Parameters
+        ----------
+        name_A, name_B: str
+            Input names of information to be mapped.
+        converters: tuple, optional, default=(str,str)
+            Converter functions applied on each entry (row) of the file
+            providing the name_a-name_B matches.
+            The order of the list elements must agree with that of the
+            object names passed in in the constructor.
+            By default, data on the file is assumed to be strings,
+            which is the typical case for particle names.
+        filename: string or file object, optional, default='particle/data/conversions.csv'.
+            Specify a file from which to read all name_a-name_B matches.
+            It is assumed that the order of items in the file matches the order
+            of arguments specified in the class constructor, hence val_A,val_B.
+
+        Examples
+        --------
+
+        >>> from particle import data  # doctest: +SKIP
+        >>> filename = data.open_text(data, 'a_to_b.csv')  # doctest: +SKIP
+        >>> A2BMap, B2AMap = DirectionalMaps('A', 'B', filename=filename)  # doctest: +SKIP
+        """
+
+        name_A = name_A.upper()
+        name_B = name_B.upper()
+
+        fieldnames = None
+        skipinitialspace = False
+
+        if filename is None:
+            filename = data.open_text(data, 'conversions.csv')
+            fieldnames = ['PDGID', 'PYTHIAID', 'GEANTID', 'PDGNAME', 'EVTGENNAME', 'LATEXNAME']
```

Comment:
This may be fine for now, but eventually it might be nice to generalize this so that it works with any file, and then these are specified when calling DirectionalMaps.

Reply:
Agreed. There is probably a bit more to streamline, now that I can make better use of the master conversions CSV file. But that will be for later, to release the fair amount of updates in the pipeline.